### ISSUE TITLE
Fix false positives for template literal interpolations in media-feature-name-no-unknown

### ISSUE THREAD

**Issue by maclockard** [stylelint/stylelint#4987]
> Clearly describe the bug

There are false positives when specifying a media query value using CSS in JS.

> Which rule, if any, is the bug related to?

`media-feature-name-no-unknown`

> What code is needed to reproduce the bug?

```js
const StyledLogo = styled.div`
  overflow: hidden;

  @media (max-width: ${({ theme }) => theme.mediaQuery.MEDIUM}px) {
    width: 18px;
  }
`;
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "media-feature-name-no-unknown": true
  }
}
```

> Which version of stylelint are you using?

`13.6.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint './**/*.{tsx,ts,css}`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, its related to using CSS in JS using styled-components

> What did you expect to happen?

No errors to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following errors were flagged:

```shell
components/app/NavigationBar.tsx
 75:11  ✖  Unexpected unknown media feature name "m"   media-feature-name-no-unknown
```

This is related to https://github.com/stylelint/stylelint/issues/4574

**Comment by maclockard** [comment]
Also tested `13.7.2`, still got the failure though

**Comment by hudochenkov** [comment]
It is more likely related to other issues with template literals interpolations listed in https://github.com/stylelint/stylelint/issues/4574. There is some work is happening to fix this in https://github.com/stylelint/stylelint/pull/4944.

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

We'll also [be deprecating](https://github.com/stylelint/postcss-css-in-js/issues/225) our [forked CSS-in-JS package](https://github.com/stylelint/postcss-css-in-js), in favour of leaner custom syntaxes.

Please consider [writing a custom syntax](https://github.com/stylelint/stylelint/blob/main/docs/developer-guide/syntaxes.md) specific to the CSS-in-JS library in this issue and fixing the bug there.
